src/QsCompiler/QirGeneration/Subtransformations/ExpressionKindTransformation.cs

```diff
@@ -386,6 +386,252 @@ IValue CopyAndUpdateUdt(TupleValue originalValue)
             return ResolvedExpressionKind.InvalidExpr;
         }
 
+        /// <summary>
+        /// Creates a string literal that contains the given values as interpolation arguments.
+        /// The given string is expected to consist of text interspaced with integer constants between curly brackets.
+        /// The content between non-escaped curly brackets is parsed as integer,
+        /// and replaced with the string representation of the interpolation argument at that index.
+        /// Registers the built string with the scope manager.
+        /// </summary>
```

Comment:
Consider adding an example (and maybe even starting with an example, such that the reader understands the most of the idea in a few seconds).
By looking just at the summary it is somewhat hard to understand, without an example, for a new/unaware reader.

Reply:
I added an example string, not sure if that's what you had in mind.